app/controllers/optIn/CheckYourAnswersController.scala

```diff
@@ -55,9 +60,28 @@ class CheckYourAnswersController @Inject()(val optInService: OptInService,
   def show(isAgent: Boolean = false): Action[AnyContent] = auth.authenticatedAction(isAgent) {
     implicit user =>
       withRecover(isAgent) {
-        optInService.fetchSavedChosenTaxYear().map { taxYear =>
-          Ok(s"Check your answers (${taxYear.get}) page! in MISUV-8006")
-        }
+
+        val result = for {
+          taxYear <- OptionT(optInService.fetchSavedChosenTaxYear())
+          cancelURL = ReportingFrequencyPageController.show(isAgent).url
+          intentIsNextYear = taxYear.isNextTaxYear(dateService)
+        } yield Ok(view(MultiYearCheckYourAnswersViewModel(taxYear, isAgent, cancelURL, intentIsNextYear)))
+
+        result.getOrElse(errorHandler(isAgent).showInternalServerError())
```

Comment:
Could we move this logic out of the controller and instead build a view model in the Service? This will make testing this logic much simpler.

Reply:
will implement this comment in this PR https://github.com/hmrc/income-tax-view-change-frontend/pull/2479/files as the same code is touched, added your comment there